docs/spec/auth/token.md

```diff
@@ -132,45 +149,51 @@ the `WWW-Authenticate` header.
-digest/challenge/response authentication scheme if the client doesn't support
-TLS client certs). If the key in the client certificate is linked to an account
-then the token is issued for that account key. If the key in the certificate is
-linked to multiple accounts then the client must specify the `account` query
-parameter. The returned token is in JWT (JSON Web Token) format, signed using
-the authorization server's private key.
-
-#### Example
+#### Example Token Request
 
-For this example, the client makes an HTTP request to the following endpoint
-over TLS using a client certificate with the server being configured to allow a
-non-verified issuer during the handshake (i.e., a self-signed client cert is
-okay).
+For this example, the client makes an HTTP request to the following URL:
 
 ```
-GET /v2/token/?service=registry.docker.com&scope=repository:samalba/my-app:push&account=jlhawn HTTP/1.1
-Host: auth.docker.com
+https://auth.docker.io/token?service=registry.docker.io&scope=repository:samalba/my-app:pull,push HTTP/1.1
 ```
 
-The server first inspects the client certificate to extract the subject key and
-lookup which account it is associated with. The client is now authenticated
-using that account.
-
-The server next searches its access control list for the account's access to
-the repository `samalba/my-app` hosted by the service `registry.docker.com`.
+The token server should first attempt to authenticate the client using any
+authentication credentials provided with the request. As of Docker 1.8, the
+registry client in the Docker Engine only supports Basic Authentication to
+these token servers. If an attempt to authenticate to the token sever fails,
+the token server should return a `401 Unauthorized` response indicating that
+the provided credentials are invalid.
+
+Whether the token server requires authentication is up to the policy of that
+access control provider. Some requested access may require authentication to
+determine access (such us pushing or pulling a private repository) while some
+access may not (such as pulling from a public repository).
+
+After authenticating the client (which may simply be an anonymous client if
+no attempt was made to authenticate), the token server must next query its
+access control list to determine whether the client has the requested scope. In
+this example request, if I have authenticated as user `jlhawn`, the token
+server will determine what access I have to the repository `samalba/my-app`
+hosted by the entity `registry.docker.io`.
+
+Once the token server has determined what access the client has to the
+resources requested in the `scope` parameter, it will take the intersection of
+the set of requested actions on each resource and the set of actions that the
+client has in fact been granted. If the client only has a subset of the
+requested access **it should not be considered an error** as it should not be
```

Comment:
'it should not be considered the responsibility' -> 'isn't considered the responsibility'